src/pages/books.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from 'react';
+import BookCard from '../components/BookCard';
+import { IBook } from '../interfaces/books';
+
+const Books = () => {
+    const [books, setBooks] = useState<IBook[]>([]);
+    const [isLoading, setIsLoading] = useState(true);
```

Comment:
Just sharing - a little trick I have been using is to have a single tristate value here -
- `undefined` (as the initial value) when loading the data
- `null` when loading fails
- The actual data (`IBook[]`) when loading succeeds

It is fine to use two `useState`, just sharing what I have been doing instead